src/pages/ProfilePage.js

```diff
@@ -112,6 +116,15 @@ function ProfilePage({ match }) {
     }
   }
 
+  let rating = null;
+  if (!loadingRatings) {
+    if (ratings == null) {
+      rating = 1200;
+    } else {
+      rating = ratings[modeVariant] || 1200;
```

Comment:
If `loadingRatings` is true, that means that `ratings` is `null` at this point. I don't think you can access  the value corresponding to `modeVariant` key here.

Reply:
`loadingRatings` is negated in the if statement, so the body of the if statement can only be reached if it has finished loading. In the body, ratings will only be `null` if the ratings object does not exist in the database yet for this user. Then we take the default case of 1200, otherwise we take the existing rating.